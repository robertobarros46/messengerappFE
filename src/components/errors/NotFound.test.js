import React from 'react';
import  { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router'
import { Link } from 'react-router-dom';
import { Button } from 'antd';


configure({ adapter: new Adapter() });

import NotFound from './NotFound';
describe('<NotFound />', () => {
    let wrapper;

    beforeEach(() => {
        wrapper = shallow(<NotFound />);
    });

    it('renders', () => {
        expect(wrapper.exists()).toBe(true);
    });

    it('should render Not found component', () => {
        const tree = renderer.create(
            <MemoryRouter>
                <NotFound />
            </MemoryRouter>).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('should have one Link and one Button components', () => {
        expect(wrapper.find(Button, Link)).toHaveLength(1);
    });

    it('test whole content', () => {
        expect(wrapper.contains(
            <div className="page-not-found">
                <h1 className="title">
                    404
                </h1>
                <div className="desc">
                    A página que você esta procurando não existe.
                </div>
                <Link to="/"><Button className="go-back-btn" type="primary" size="large">Go Back</Button></Link>
            </div>
        )).toEqual(true);
    });
})