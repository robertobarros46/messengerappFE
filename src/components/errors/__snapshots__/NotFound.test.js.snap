// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NotFound /> should render Not found component 1`] = `
<div
  className="page-not-found"
>
  <h1
    className="title"
  >
    404
  </h1>
  <div
    className="desc"
  >
    A página que você esta procurando não existe.
  </div>
  <a
    href="/"
    onClick={[Function]}
  >
    <button
      className="ant-btn go-back-btn ant-btn-primary ant-btn-lg"
      onClick={[Function]}
      type="button"
    >
      <span>
        Go Back
      </span>
    </button>
  </a>
</div>
`;
